assets/js/search.js

```diff
@@ -0,0 +1,42 @@
+var loadedMiniSearch;
+
+async function loadMiniSearch() {
+  if (loadedMiniSearch === undefined) {
+    const response = await fetch('/search-index.json');
+    const docs = await response.json();
+    loadedMiniSearch = new MiniSearch({
+      // 'id' is implicitly added to fields and storeFields
+      fields: ['title', 'text'],
+      storeFields: ['title'],
+      searchOptions: { 'prefix': true }
```

Comment:
I tried it and it gave waaaay too many results. I think I'll leave this to a follow-up PR since it likely involves spending time tweaking coefficients for how fuzzy, how many characters til we look for results, etc.